TestFoundation/TestStream.swift

```diff
@@ -208,6 +257,42 @@ class TestStream : XCTestCase {
         XCTAssertEqual(.error, outputStream!.streamStatus)
     }
     
+    func test_outputStreamDelegate() {
+        guard let filePath = createTestFile("TestFileOut_delegate.txt", _contents: Data(capacity: 256)) else {
+            XCTFail("Unable to create temp file");
+            return
+        }
+        
+        guard let outputStream = OutputStream(url: URL(fileURLWithPath: filePath), append: true) else {
+            XCTFail("Unable to create output stream");
+            return
+        }
+        XCTAssertEqual(.notOpen, outputStream.streamStatus)
+        var myString = "Hello world!"
+        let encodedData = [UInt8](myString.utf8)
+        
+        let delegate = TestStreamDelegate.init() { eventCode in
+            switch eventCode {
+            case .openCompleted:
+                XCTAssertEqual(.open, outputStream.streamStatus)
+                expectation.fulfill()
+            case .hasSpaceAvailable:
+                expectation.fulfill()
+            case .errorOccurred:
+                let errorDesc = outputStream.streamError?.localizedDescription ?? "No description"
+                XCTFail("Output stream delegate test error occured: \(errorDesc)");
+            default:
+                break
+            }
+        }
+        outputStream.delegate = delegate
+        outputStream.schedule(in: RunLoop.main, forMode: .defaultRunLoopMode)
+        outputStream.open()
+        let result = outputStream.write(encodedData, maxLength: encodedData.count)
+        outputStream.close()
+        removeTestFile(filePath)
```

Comment:
Do we not have to spin the run loop here?

Reply:
I've forgot to add `wait(for: [expectation], timeout: 10)` here as I did for input stream. If you have better way to handle this, I'm eager to learn, otherwise we can add that wait function